src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -1005,8 +1129,15 @@ ss::future<> ntp_archiver::upload_until_term_change_legacy() {
             // grow very large disabling the archival storage
             vlog(
               _rtclog.trace, "Nothing to upload, applying backoff algorithm");
-            co_await _wakeup_event.wait(
-              backoff + _backoff_jitter.next_jitter_duration());
+            auto timeout = backoff + _backoff_jitter.next_jitter_duration();
+            // Reset the context before sleeping to avoid holding to the memory.
+            // Most of the time the archiver spends in the following 'wait'
+            // call. It's not possible for it to get stalled there so it's safe
+            // to reset the context here. This guarantees that we only consuming
+            // additional memory for traces for the duration of the upload.
```

Comment:
> consuming additional memory for traces

what stops us from accumulating trace logs while the context is suspended? or is this referring to the overhead of the actual trace dump?

Reply:
actually, i guess this is a comment on releasing already produced traces, yeah?